### ISSUE TITLE
Adding option for custom padding

### ISSUE THREAD

**Issue by AdityaKane2001** [keras-team/keras#15504]
**System information**.

TensorFlow version (you are using): 2.7
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**.

Currently the Conv2D class, for example, accepts padding="same" or "valid". There is no other option for anyone who wants custom padding. I propose to add functionality such that `padding=<any format which ZeroPadding accepts>`. This is useful because one does not have to worry about incorrect padding or adding an unnecessary layer in the model. We can handle this internally with tf.pad. This is also similar as in PyTorch.   

**Will this change the current api? How?**

Conv{1,2,3}D will accept integers or tuple of integers or tuple of tuple of integers as valid arguments.

**Who will benefit from this feature?**

The Keras community.

/cc @haifeng-jin @fchollet

**Comment by deeb02** [comment]
Seems to be achievable through the [ZeroPadding2D](https://keras.io/api/layers/reshaping_layers/zero_padding2d/) layer. Adding @fchollet in case he wanted to add some details.

**Comment by AdityaKane2001** [comment]
@deeb02 

Should open a PR so we can discuss there?

/cc @fchollet
